src/opm/parser/eclipse/EclipseState/Aquancon.cpp

```diff
@@ -113,10 +124,60 @@ namespace Opm {
     }
 
 
+    bool Aquancon::cellActive(const Opm::EclipseGrid& grid, const int i, const int j, const int k)
+    {
+        if ( i <= 0 || j <=0 || k <=0
+            || size_t(i) > grid.getNX()
+            || size_t(j) > grid.getNY()
+            || size_t(k) > grid.getNZ())
```

Comment:
Is indices outside the grid box at all legal? My initial reaction would be just to throw an exception in this case (rendering the whole `Aquancon::cellActive()` method obsolete).

Reply:
> Is indices outside the grid box at all legal? 

No. That is why we can not use the `cellActive()` from the EclipseGrid.  Basically the point here is that the Aquifer by default is located at the face of the outer boundary of the reservoir, not inside the reservoir. 

But you brought up a good point, it might be able to be done in a different way.  Thanks for your comments, I will try to address them later. 

It is marked as WIP is because we has not get everything correct with the targeting case, we are waiting for the reference output to help to figure out the more complete logic here.

Reply:
I am just leaving a note for later reference for my self. 

`cellActive()` is a little confusing name, and it is used for other purpose. A function name indicating whether a cell is inside or outside the reservoir is better.